examples/run/run.cpp

```diff
@@ -694,9 +694,9 @@ class LlamaData {
             return 1;
         }
 
-        const std::string org          = repo_parts[0];
-        const std::string project      = repo_parts[1];
-        std::string       project_path = repo_parts[2];
+        const std::string & org          = repo_parts[0];
```

Comment:
Some small optimisations I spotted yesterday, less copies.

Reply:
IMO you can shorten the code to generate `repo_parts` by abstracting some logic into a new function `string_split`

Or you can reuse the `string_split<std::string>` in common.h

Btw maybe also validate the length of `repo_parts` to avoid accessing out-of-bound

Reply:
length validation is here:

```
   if (repo_parts.size() < 3) {
```

Reply:
Used string_split